### PR TITLE
feat(Dropdown): [SPT-1853] add disable popper prop

### PR DESCRIPTION
**Pull request by 13ANM**
[FX-1853](https://toptal-core.atlassian.net/browse/SPT-1853)

### Description

Added `disablePopper` option to fix overflow issues in SP

### How to test

`/?path=/story/components-dropdown--dropdown#positions-offsets`

### Screenshots

https://user-images.githubusercontent.com/79158829/129924658-c6f8db61-fa1b-4046-9085-45de18c14e22.mov

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Review by 13ANM — COMMENTED** [review]
Need to fix: 

image

Happens when scrolling the page with the added example open.

File: packages/picasso/src/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -240,7 +243,14 @@ export const Dropdown = forwardRef<HTMLDivElement, Props>(function Dropdown (
             When the anchor goes above the viewport, popper goes to infinite flipping.
             flipped: true -> flipped: false -> flipped: true -> ...
             */
-            modifiers: { flip: { enabled: contentOverflow !== 'visible' } },
+            modifiers: {
+              flip: {
+                enabled: !disablePopper && contentOverflow !== 'visible'
```
Comment (OleksandrNechai) [inline packages/picasso/src/Dropdown/Dropdown.tsx:248]: does disabling flip disables the whole popper?

File: packages/picasso/src/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -49,6 +49,8 @@ export interface Props extends StandardProps, HTMLAttributes<HTMLDivElement> {
   disableAutoClose?: boolean
   /** Disable the portal behavior. The children stay within it's parent DOM hierarchy. */
   disablePortal?: boolean
+  /** Disable the popper behavior. The dropdown will remain fixed below the button. */
+  disablePopper?: boolean
```
Comment (OleksandrNechai) [inline packages/picasso/src/Dropdown/Dropdown.tsx:53]: From the description, I am not sure why are we doing that? So do I understand correctly in SP we have trouble with the wrong zIndex of overlays?

File: packages/picasso/src/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (13ANM) [reply packages/picasso/src/Dropdown/Dropdown.tsx:248]: No, just the flip. We have one case in which we want to have flip disabled but not preventOverflow, when contentOverflow is 'visible'

File: packages/picasso/src/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (13ANM) [reply packages/picasso/src/Dropdown/Dropdown.tsx:53]: We want to follow old platform and have the dropdown remain fixed below the button

File: packages/picasso/src/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (OleksandrNechai) [reply packages/picasso/src/Dropdown/Dropdown.tsx:248]: This problem is not trivial. We have a `popperOptions` here and we need to use it to control the Popper behavior. I have talked to @havenchyk about this problem and we discussed some options. 
@havenchyk could you please after you have thought it through help pushing this forward? :bow:

File: packages/picasso/src/Dropdown/story/PositionsAndOffsets.example.tsx [hunk 3]
```diff
@@ -53,6 +53,22 @@ const Example = () => (
           <Dropdown.Arrow />
         </Dropdown>
       </Container>
+
+      <Container right='large'>
```
Comment (OleksandrNechai) [inline packages/picasso/src/Dropdown/story/PositionsAndOffsets.example.tsx:57]: I think there is a little value in demonstrating different Popper options as there are too many of them

File: packages/picasso/src/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (OleksandrNechai) [reply packages/picasso/src/Dropdown/Dropdown.tsx:248]: another option. @havenchyk maybe we could encapsulate the original `popperOptions` and have instead popperOptions specialized to this component:
```
{
   flip:boolean
   contentOverflow: 'visible' | 'scroll'
}
```
later we transform this :point_up: simple object into complicated `popperOptions` that Popper understands. The approach is less flexible but has a simple interface which I guess is a good thing.

**Comment by 13ANM** [comment]
Closed as the new prop is not necessary, `popperOptions` can be used for this case.

**Pull request closed without merging**